File: Python/power4.py
# https://leetcode.com/problems/power-of-four/description/

class Solution:
    def power4(self, n: int) -> bool:
        return n > 0 and (n & (n - 1)) == 0 and (n & 0x55555555) != 0
    
if __name__ == "__main__":
    sol = Solution()
    test_values = [1, 4, 16, 5, 0, -4, 64, 100]
    for val in test_values:
        print(f"Is {val} a power of 4? {sol.power4(val)}")

File: power4.java
// https://leetcode.com/problems/power-of-four/description/

class power4{
    public boolean isPowerOfFour(int n) {
        return n > 0 && (n & (n - 1)) == 0 && (n & 0x55555555) != 0;
    }

    public static void main(String[] args) {
        power4 p4 = new power4();
        System.out.println(p4.isPowerOfFour(16)); // true
        System.out.println(p4.isPowerOfFour(5));  // false
    }
}